src/game/PowerManager.cxx

```diff
@@ -156,4 +136,44 @@ std::vector<PowerGrid> PowerManager::rebuildZoneArea(PowerGrid &powerGrid)
   }
 
   return newPowerGrids;
+}
+void PowerManager::updatePlacedNodes(const MapNode &mapNode)
+{
+  TileData *tileData; // we need tileData from conductive tiles (buildings or powerlines)
+  if (mapNode.getTileData(Layer::BUILDINGS))
+    tileData = mapNode.getTileData(Layer::BUILDINGS);
+  else if (mapNode.getTileData(Layer::POWERLINES))
+    tileData = mapNode.getTileData(Layer::POWERLINES);
+  else if (mapNode.getTileData(Layer::ZONE))
+    tileData = mapNode.getTileData(Layer::ZONE);
+  else
+    return;
+
+  PowerNode nodeToAdd = {mapNode.getCoordinates(), tileData->power};
+  m_nodesToAdd.push_back(nodeToAdd);
+}
+
+void PowerManager::updateRemovedNodes(const MapNode *mapNode)
+{
+  switch (GameStates::instance().demolishMode)
+  {
+  case DemolishMode::DEFAULT:
+  {
+    if (!mapNode->getTileData(Layer::BUILDINGS))
+    { // if there's no building it has been succesfully demolished
+      m_nodesToRemove.push_back(mapNode->getCoordinates());
```

Comment:
IMHO, this may be comfusing here: When buildings in a zone has been partiallly demolised which may split the powergrid into two seperate powergrids. But the remained buildings are still in a zone. This may bother the players to some extent.

Here is an example: 
![image](https://user-images.githubusercontent.com/41318515/168411781-381b36b0-306d-429a-b521-84128955d6dc.png)
the buildings in the zone will not spawn even though the zone is adjacent to a powerplant.

Reply:
This is a real issue (and a good find). Powernodes can be duplicated entries in the m_powerNodes vector (wich are deleted by coordinates as you pointed out). Determining the node by power level is also not distinguishable enough. I might add the layer here to determine what tile has been demolished. This is now error prone (as only one object per coordinate would get deleted).

I will also check if there's no zone tile on the same layer and look at the code again. Thank you!

Reply:
fixed